### PR TITLE
T5725: Improve protocol IS-IS config validation

### PR DESCRIPTION
**Pull request by HollyGurza**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T5725

## How to test

```
set interfaces dummy dum0 address '203.0.113.254/32'
set interfaces dummy dum0 description 'dum0'
set interfaces ethernet eth1 address '192.0.2.5/24'
set interfaces ethernet eth1 description 'LAN'

set interfaces tunnel tun0 address '10.0.0.2/30'
set interfaces tunnel tun0 description 'tun-to-192.0.2.1'
set interfaces tunnel tun0 encapsulation 'gre'
set interfaces tunnel tun0 source-address '192.0.2.5'

set protocols isis interface eth1
set protocols isis interface tun0
set protocols isis lsp-mtu '1460'
set protocols isis net '49.0001.1920.0200.0011.00'
set protocols isis redistribute ipv4 connected level-2
commit
# here we get error
# Option remote for interface tun0 is required.

set interfaces tunnel tun0 remote '192.0.2.1'
commit
# commit is success
```
check frr config:
```
vyos@vyos# vtysh -c "show run isisd"
Building configuration...

Current configuration:
!
frr version 7.5.1-20240509-02-gc9dd2977e
frr defaults traditional
hostname vyos
log syslog
log facility local7
service integrated-vtysh-config
!
interface eth1
 ip router isis VyOS
 ipv6 router isis VyOS
!
interface tun0
 ip router isis VyOS
 ipv6 router isis VyOS
!
router isis VyOS
 net 49.0001.1920.0200.0011.00
 lsp-mtu 1460
 redistribute ipv4 connected level-2
!
line vty
!
end
```
## Smoketest result

```
vyos@vyos:~$ python3 /usr/libexec/vyos/tests/smoke/cli/test_protocols_isis.py 
test_isis_01_redistribute (__main__.TestProtocolsISIS) ... ok
test_isis_02_zebra_route_map (__main__.TestProtocolsISIS) ... ok
test_isis_03_default_information (__main__.TestProtocolsISIS) ... ok
test_isis_04_password (__main__.TestProtocolsISIS) ... 
Can use either md5 or plaintext-password for area-password!


Can use either md5 or plaintext-password for domain-password!

ok
test_isis_06_spf_delay (__main__.TestProtocolsISIS) ... 
All types of delay must be specified: long-delay, time-to-learn, init-
delay, short-delay


All types of delay must be specified: long-delay, time-to-learn, short-
delay


All types of delay must be specified: time-to-learn, short-delay


All types of delay must be specified: time-to-learn

ok
test_isis_06_tunnel_interface (__main__.TestProtocolsISIS) ... 
Option remote for interface tun0 is required.

ok

----------------------------------------------------------------------
Ran 6 tests in 39.461s

OK
```
## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by c-po** [comment]
This check is not present on `current` it should/must be implemented there first end then backported to `sagitta` and `equuleus` branch

File: src/conf_mode/protocols_isis.py [hunk 1]
```diff
@@ -69,6 +69,15 @@ def get_config(config=None):
     # Merge policy dict into "regular" config dict
     isis = dict_merge(tmp, isis)
 
+    for interface in isis.get('interface'):
+        # when we use tunnel interface necessary additional config validate
+        if 'tun' in interface:
```
Comment (c-po) [inline src/conf_mode/protocols_isis.py]: Please use: `if interface.startswith('tun')` instead.

File: src/conf_mode/protocols_isis.py [hunk 2]
```diff
@@ -103,6 +112,10 @@ def verify(isis):
                               f'Recommended area lsp-mtu {recom_area_mtu} or less ' \
                               '(calculated on MTU size).')
 
+        if 'tun' in interface:
```
Comment (c-po) [inline src/conf_mode/protocols_isis.py]: Please use: `if interface.startswith('tun')` instead.

File: src/conf_mode/protocols_isis.py [hunk 3]
```diff
@@ -103,6 +112,10 @@ def verify(isis):
...
+            if not isis.get('tunnel_config').get(interface).get('remote'):
```
Comment (c-po) [inline src/conf_mode/protocols_isis.py]: Please stick to the common pattern and use `vyos.utils.dict_search()`

**Review by c-po — CHANGES_REQUESTED** [review]

**Comment by HollyGurza** [comment]
> This check is not present on `current` it should/must be implemented there first end then backported to `sagitta` and `equuleus` branch

this problem reproduces only in 1.3 version, current and sagitta works fine without this check.

File: src/conf_mode/protocols_isis.py (hunk 3 above)
Comment (HollyGurza) [reply src/conf_mode/protocols_isis.py]: done

File: src/conf_mode/protocols_isis.py (hunk 2 above)
Comment (HollyGurza) [reply src/conf_mode/protocols_isis.py]: done

File: src/conf_mode/protocols_isis.py (hunk 1 above)
Comment (HollyGurza) [reply src/conf_mode/protocols_isis.py]: done

**Review by dmbaturin — APPROVED** [review]

**Review by c-po — APPROVED** [review]
Bug only present with FRR 7.5.

Working with FRR 9.1+

**Pull request merged**